components/brave_vpn/common/wireguard/wireguard_utils.cc

```diff
@@ -48,6 +51,64 @@ constexpr char kWireguardConfigTemplate[] = R"(
   Endpoint = {vpn_server_hostname}:51821
 )";
 
+bool IsValidConfigDNS(const std::string& config_str) {
+  int match_count = 0, any_match_count = 0;
+
+  // Match our expected `DNS` field.
+  absl::string_view input(config_str);
+  while (match_count < 2 &&
+         re2::RE2::FindAndConsume(&input, R"((?m)^\s*DNS = 1\.1\.1\.1$)")) {
+    match_count++;
+  }
+
+  // Match any `DNS` field.
+  absl::string_view input_any(config_str);
+  while (any_match_count < 2 &&
+         re2::RE2::FindAndConsume(&input_any, R"((?m)^\s*DNS\s*=\s*.*$)")) {
+    any_match_count++;
+  }
+
+  if (any_match_count == 1 && match_count == 1) {
+    return true;
+  }
+
+  VLOG(1) << "Invalid configuration: " << any_match_count
+          << " (or more) `DNS` fields found, " << match_count
+          << " matching our expected `DNS` pattern.";
+  return false;
+}
+
+bool IsValidConfigEndpoint(const std::string& config_str) {
+  int match_count = 0, any_match_count = 0;
+
+  // Match our expected `Endpoint` field.
+  absl::string_view input(config_str);
+  while (
+      match_count < 2 &&
+      re2::RE2::FindAndConsume(
+          &input,
+          R"((?m)^\s*Endpoint = [^.]+\.(guardianapp\.com|sudosecuritygroup\.com):51821$)")) {
+    match_count++;
+  }
+
+  // Match any `Endpoint` field.
+  absl::string_view input_any(config_str);
+  while (
+      any_match_count < 2 &&
+      re2::RE2::FindAndConsume(&input_any, R"((?m)^\s*Endpoint\s*=\s*.*$)")) {
```

Comment:
maybe should make this a case insensitive match unless we are sure wireguard configs are case sensitive?

Reply:
same comment as above 😄